app/src/stores/post.spec.ts

```diff
@@ -123,4 +124,150 @@ describe("Post store", () => {
         expect(res1!.length).toBe(1);
         expect(res1![0].content[0].publishDate).toBe(2);
     });
+
+    it("shows posts with publish date < now and no expiry date", async () => {
+        const now = DateTime.now().toMillis();
+        const postStore = usePostStore();
+        const post1 = {
+            ...mockPost,
+            _id: "1",
+            content: [{ ...mockPost.content[0], title: "post1", publishDate: now - 1000 }],
+        } as any;
+        const post2 = {
+            ...mockPost,
+            _id: "2",
+            content: [{ ...mockPost.content[0], title: "post2", publishDate: now + 1000 }],
+        } as any;
+
+        postStore.posts = [post1, post2];
+
+        const visiblePosts = postStore.posts.filter((post) => {
+            const { publishDate, expiryDate } = post.content[0];
+            return publishDate < now && !expiryDate;
+        });
```

Comment:
I don't think that `filter` should be here, otherwise you're not testing the logic in the store.

(so the code becomes obsolete, but if you would want to work around the `publishDate is possibly undefined` error you could first check that is not, like `return publishDate && publishDate < now && !expiryDate`)

Reply:
OK ! What Logic should i test in store ? PostByTag ?